src/v/schema/registry.cc

```diff
@@ -87,6 +100,33 @@ class disabled_schema_registry : public registry {
 };
 } // namespace
 
+ss::future<ppsr::valid_schema>
+registry::get_valid_schema(ppsr::schema_id schema_id) const {
+    auto reader = co_await getter();
+    auto schema_def = co_await reader->get_schema_definition(schema_id);
```

Comment:
Yea this is a good observation, and one that I handled. That said, it's handled today as a transient error, since AFAICT there's no distinguishing a transient error and something more deterministic like missing schema (which frankly probably means the magic byte was a coincident and there is no schema). I think ideally this code would like the registry to differentiate between transient errors, but for now they are just treated the same

Reply:
yeah iirc those exceptions are meant to unwind all the way back out to a rest handler, so maybe it would be preferable to have a secondary API on schema_getter to expose the `outcome` to internal clients. anyway, thanks for the explanation.